business/workloads.go

```diff
@@ -556,6 +564,103 @@ func parseLogLine(line string, isProxy bool, engardeParser *parser.Parser) *LogE
 	}
 
 	// override the timestamp with a simpler format
+	timestamp := parseTimestamp(parsedTimestamp)
+	entry.Timestamp = timestamp
+	entry.TimestampUnix = parsedTimestamp.UnixMilli()
+
+	return &entry
+}
+
+func parseZtunnelLine(line string) *LogEntry {
+	entry := LogEntry{
+		Message:       "",
+		Timestamp:     "",
+		TimestampUnix: 0,
+		Severity:      "INFO",
+	}
+
+	splitted := strings.SplitN(line, " ", 2)
+	if len(splitted) != 2 {
+		log.Debugf("Skipping unexpected log line [%s]", line)
+		return nil
+	}
+
+	msgSplit := strings.Split(line, "\t")
+
+	if len(msgSplit) < 4 {
+		log.Debugf("Error splitting line line [%s]", line)
+		entry.Message = line
+		return &entry
+	}
+
+	entry.Message = msgSplit[4]
+	if entry.Message == "" {
+		log.Debugf("Skipping empty log line [%s]", line)
+		entry.Message = line
+		return &entry
+	}
+
+	// k8s promises RFC3339 or RFC3339Nano timestamp, ensure RFC3339
+	// Split by blanks, to get the milliseconds for sorting, try RFC3339Nano
+	ts := strings.Split(msgSplit[0], " ") // Sometime timestamp is duplicated
+	entry.Timestamp = ts[0]
+
+	// If we are past the requested time window then stop processing
+	parsedTimestamp, err := time.Parse(time.RFC3339Nano, entry.Timestamp)
+	entry.OriginalTime = parsedTimestamp
+	if err != nil {
+		log.Debugf("Failed to parse log timestamp (skipping) [%s], %s", entry.Timestamp, err.Error())
+		return nil
+	}
+
+	if splitted[1] != "" {
+		entry.Severity = strings.ToUpper(splitted[1])
+	}
+
+	// override the timestamp with a simpler format
+	timestamp := parseTimestamp(parsedTimestamp)
+	entry.Timestamp = timestamp
+	entry.TimestampUnix = parsedTimestamp.UnixMilli()
+
+	// Process some access log data
+	// More validations can be done. Data is in format direction=outbound
+	// Also, more data could be added?
+	al := parser.AccessLog{}
+	al.Timestamp = timestamp
+	if len(msgSplit) > 4 {
+		accessLog := strings.Split(msgSplit[4], " ")
+		for _, field := range accessLog {
+			parsed := strings.Split(field, "=")
```

Comment:
If one of the values can have an `=` in it (possibly `error`?) then this might be problematic.

Reply:
I've changed for `strings.SplitN(field, "=", 2)`. It is less probable that = is in the left side, so at least just the first "=" is split.